src/main/java/mil/dds/anet/threads/AnetEmailWorker.java

```diff
@@ -126,22 +124,34 @@ private void runInternal() {
 		//Send the emails!
 		final List<Integer> processedEmails = new LinkedList<Integer>();
 		for (final AnetEmail email : emails) {
-			if (this.nbOfHoursForStaleEmails != null && email.getCreatedAt().isBefore(Instant.now().atZone(DaoUtils.getDefaultZoneId()).minusHours(nbOfHoursForStaleEmails).toInstant())) {
-				logger.info("Purging stale email to {} re: {}", email.getToAddresses(), email.getAction().getSubject());
+			Map<String,Object> context = null;
+
+			try {
+				context = buildContext(email);
+				logger.info("{} Sending email to {} re: {}", disabled ? "[Disabled] " : "" , email.getToAddresses(), email.getAction().getSubject(context));
+
+				if (!disabled)
+				{
+					sendEmail(email, context);
+				}
+
 				processedEmails.add(email.getId());
-			}
-			else {
-				try {
-					if (disabled) {
-						logger.info("Disabled, not sending email to {} re: {}", email.getToAddresses(), email.getAction().getSubject());
-					} else {
-						logger.info("Sending email to {} re: {}", email.getToAddresses(), email.getAction().getSubject());
-						sendEmail(email);
+			} catch (Throwable t) {
+				//This email will never complete, just kill it.
```

Comment:
This disregards the comment in `sendEmail`:
```
		//Other errors are intentially thrown, as we want ANET to try again.
```
so transient errors now no longer result in a re-try next time around.

Reply:
yes they will, If sendEmail throws an exception, `processedEmails.add(email.getId());` won't be called and that email will be processed next time

Reply:
Then please update the comment
```
				//This email will never complete, just kill it.
```